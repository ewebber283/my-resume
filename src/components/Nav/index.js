import React from 'react';

function Nav() {

function categorySelected(name) {
    console.log(`${name}`)
    }
const categories = [
    { name: "Projects"},
    { name: "Experience" },
];
return (
    <header>
        <h2>
    <a href="/">
      Eric Webber
    </a>
  </h2>
        <nav>
            <ul className="flex-row">
                <li className="mx-2">
                    <a href="#about">
                    About me
                    </a>
                </li>
                <li>
                    <span>Contact</span>
                </li>
                {categories.map((category) => (
                    <li
                        className='mx-1'
                        key={category.name}
                        >
                            <span onClick={() => categorySelected(category.name)} >
                                {category.name}
                            </span>
                        </li>
                ))}
            </ul>
        </nav>
    </header>
  );
}

export default Nav;