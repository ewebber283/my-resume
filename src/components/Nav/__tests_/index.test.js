import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav component', () => {
    // baseline test
    it('renders', () => {
      render(<Nav />);
    });
  
    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Nav />);
      
        expect(asFragment()).toMatchSnapshot();
      });
      
  })

  describe('links are visible', () => {
    it('inserts text into the links', () => {
      const { getByTestId } = render(<Nav />);
  
      expect(getByTestId('link')).toHaveTextContent('Eric Webber');
      expect(getByTestId('about')).toHaveTextContent('About me');
    });
  
  })