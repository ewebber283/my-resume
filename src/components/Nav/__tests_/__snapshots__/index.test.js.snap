// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        Eric Webber
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Projects
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Experience
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
